src/Microsoft.OData.Edm/Vocabularies/CommunityVocabularyModel.cs

```diff
@@ -44,32 +44,67 @@ public static class AlternateKeysVocabularyModel
         internal static readonly IEdmComplexType PropertyRefType;
 
         /// <summary>
-        /// Parse Alternate Keys Vocabulary Model from AlternateKeysVocabularies.xml
+        /// Parse Alternate Keys Vocabulary Model from CommunityVocabularies.xml
         /// </summary>
         static AlternateKeysVocabularyModel()
+        {

```

Comment:
From the C# spec:

A static constructor is called automatically to initialize the class before the first instance is created or any static members are referenced.

I am wondering about the order of execution about the static field "Instance" and the static constructor. For example:

If I access the "AlternateKeysVocabularyModel.Instance", before any code, what will happen?
If static constructor executed before field access, "Instance" is a null reference.

Maybe I am wrong, please make sure the calling is in correct order.

Reply:
As we checked off line, the AlternateKeysVocabularyModel.Instance could be a null reference depends on initialization order of static intances / fields. But when it is null, it will be initialized first before it's used. So the static field should not be null when it is used.